pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -461,10 +461,15 @@ func (c *Controller) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 			waitUntilExtensionResourcesDeleted,
 			waitUntilContainerRuntimeResourcesDeleted,
 		)
+		destroyKubeAPIServerSNI = g.Add(flow.Task{
+			Name:         "Destroying kube-apiserver-sni",
+			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerSNI.Destroy).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(syncPointCleaned),
+		})
 		destroyControlPlane = g.Add(flow.Task{
 			Name:         "Destroying shoot control plane",
 			Fn:           flow.TaskFn(botanist.Shoot.Components.Extensions.ControlPlane.Destroy).RetryUntilTimeout(defaultInterval, defaultTimeout),
-			Dependencies: flow.NewTaskIDs(syncPointCleaned),
+			Dependencies: flow.NewTaskIDs(syncPointCleaned, destroyKubeAPIServerSNI),
```

Comment:
Why the `destroyKubeAPIServerSNI` step is executed right before `destroyControlPlane`? What is the dependency between them? Why for example `destroyKubeAPIServerSNI` is not executed in parallel with `deleteKubeAPIServer`? Does this really work, can the ControlPlane be deleted when the `kube-apiserver-sni` stuffs are removed?

Reply:
SNI depends on kube-apiserver Service to be there. When we create the cluster we first create the kube-apiservice then SNI. Deletion should be in reversed order - SNI, then kube-apiserver service.

Reply:
Can you point me where the kube-apiserver Service is being deleted explicitly?

Reply:
> Does this really work, can the ControlPlane be deleted when the kube-apiserver-sni stuffs are removed?

This probably works as I didn't consider that the communication to the apiserver is in-cluster.

Reply:
Outdated. `kube-apiserver-sni` and `kube-apiserver-service` are now explicitly deleted after the `kube-apiserver`.